pkg/kubelet/cm/dra/plugin/client.go

```diff
@@ -81,126 +62,114 @@ func NewDRAPluginClient(pluginName string) (drapb.NodeClient, error) {
-					})
-				result := &drapb.NodePrepareResourceResponse{}
-				if errOld != nil {
-					result.Error = errOld.Error()
-				} else {
-					result.CDIDevices = respOld.CdiDevices
-				}
-				resp.Claims[claim.Uid] = result
+	if r.pluginAPIVersion == "v1alpha3" {
+		nodeClient := drapbv1alpha3.NewNodeClient(conn)
+		response, err = nodeClient.NodePrepareResources(ctx, req)
+		return
+	}
+
+	if r.pluginAPIVersion == "v1alpha2" {
+		nodeClient := drapbv1alpha2.NewNodeClient(conn)
+		for _, claim := range req.Claims {
+			res, err := nodeClient.NodePrepareResource(ctx, convertV1alpha3ClaimToV1alpha2NodePrepareResourceRequest(claim))
+			if err != nil {
+				return nil, fmt.Errorf("error calling v1alpha2.NodePrepareResource: %v", err)
 			}
+			response.Claims[claim.Uid] = convertV1alpha2NodePrepareResourceResponseToV1alpha3NodePrepareResourceResponse(res)
 		}
+		return
 	}
 
-	return
+	return nil, fmt.Errorf("unknown plugin API version")
 }
 
 func (r *draPluginClient) NodeUnprepareResources(
 	ctx context.Context,
-	req *drapb.NodeUnprepareResourcesRequest,
+	req *drapbv1alpha3.NodeUnprepareResourcesRequest,
 	opts ...grpc.CallOption,
-) (resp *drapb.NodeUnprepareResourcesResponse, err error) {
+) (response *drapbv1alpha3.NodeUnprepareResourcesResponse, err error) {
 	logger := klog.FromContext(ctx)
 	logger.V(4).Info(log("calling NodeUnprepareResource rpc"), "request", req)
-	defer logger.V(4).Info(log("done calling NodeUnprepareResources rpc"), "response", resp, "err", err)
+	defer logger.V(4).Info(log("done calling NodeUnprepareResources rpc"), "response", response, "err", err)
 
-	if r.nodeClientCreator == nil {
-		return nil, errors.New("failed to call NodeUnprepareResources. nodeClientCreator is nil")
-	}
-
-	nodeClient, nodeClientOld, closer, err := r.nodeClientCreator(r.addr)
+	conn, err := r.plugin.getOrCreateGRPCConn()
 	if err != nil {
 		return nil, err
 	}
-	defer closer.Close()
 
 	ctx, cancel := context.WithTimeout(ctx, PluginClientTimeout)
 	defer cancel()
 
-	resp, err = nodeClient.NodeUnprepareResources(ctx, req)
-	if err != nil {
-		status, _ := grpcstatus.FromError(err)
-		if status.Code() == grpccodes.Unimplemented {
-			// Fall back to the older gRPC API.
-			resp = &drapb.NodeUnprepareResourcesResponse{
-				Claims: make(map[string]*drapb.NodeUnprepareResourceResponse),
-			}
-			err = nil
-			for _, claim := range req.Claims {
-				_, errOld := nodeClientOld.NodeUnprepareResource(ctx,
-					&drapbv1alpha2.NodeUnprepareResourceRequest{
-						Namespace:      claim.Namespace,
-						ClaimUid:       claim.Uid,
-						ClaimName:      claim.Name,
-						ResourceHandle: claim.ResourceHandle,
-					})
-				result := &drapb.NodeUnprepareResourceResponse{}
-				if errOld != nil {
-					result.Error = errOld.Error()
-				}
-				resp.Claims[claim.Uid] = result
+	if r.pluginAPIVersion == "v1alpha3" {
+		nodeClient := drapbv1alpha3.NewNodeClient(conn)
+		response, err = nodeClient.NodeUnprepareResources(ctx, req)
+		return
```

Comment:
Ditto here

Reply:
done in 38653c477e7590234a92dc58d9fb73a1c7a6f385